src/elf/mod.rs

```diff
@@ -348,18 +348,18 @@ if_sylvan! {
 
     fn gnu_hash_len(bytes: &[u8], offset: usize, ctx: Ctx) -> error::Result<usize> {
         let buckets_num = bytes.pread_with::<u32>(offset, ctx.le)? as usize;
-        let min_chain = bytes.pread_with::<u32>(offset + 4, ctx.le)? as usize;
-        let bloom_size = bytes.pread_with::<u32>(offset + 8, ctx.le)? as usize;
+        let min_chain = bytes.pread_with::<u32>(offset.wrapping_add(4), ctx.le)? as usize;
```

Comment:
ok back to this PR; so what are the semantics if this overflows? This seems like a subtle bug now (I guess it was always present) Also, won't this only crash in debug mode, but will wrap overflow in release? Ditto for the other explicit overflow wraps

Reply:
I'm using `wrapping_xxx` here because I've seen other code do it too, for example, 
https://github.com/m4b/goblin/blob/2188699dcc242bae0e58b052b029521b4c8a100a/src/mach/imports.rs#L234, semantically, maybe `overflowing_xxx` is better?

Reply:
yes, i think your intuition is right, was kind of just wondering out loud what pros/cons are of the explict wrap, and what it means for the gnu hash table calculations to overflow (i just assume it means the binary is bad, better not to crash then i suppose?)

Reply:
Does `gnu_hash_len` calculate total size of GNU_HASH table?
Then performing some checks like https://github.com/m4b/goblin/blob/ece3ce60fbe5e7e6f31c69fd462d1b37521b0dc8/src/elf/gnu_hash.rs#L116-L135 is Ok.